internal/pkg/platform/deploy.go

```diff
@@ -223,7 +223,7 @@ func packageArchive(ctx context.Context, clients *shared.ClientFactory, projectR
 		DstDirPath: tmpDir,
 		AuthTokens: authTokens,
 	}
-	if err := clients.Runtime.PreparePackage(clients.SDKConfig, clients.HookExecutor, preparePackageOpts); err != nil {
+	if err := clients.Runtime.PreparePackage(ctx, clients.SDKConfig, clients.HookExecutor, preparePackageOpts); err != nil {
```

Comment:
note: We also needed to update the `package runtime` interface for `PreparePackage` to accept a `ctx` argument. On the upside, this aligns with the other functions in the interface, which also accept a `ctx`.